From 8d7a3f40c8de414122eb8eb2131291782ee8ed15 Mon Sep 17 00:00:00 2001
From: Daniel Sabogal <dsabogalcc@gmail.com>
Date: Fri, 23 Sep 2016 21:10:07 -0400
Subject: [PATCH 12/18] fix getservby*_r result pointer value on error

this is a clone of the fix to the gethostby*_r functions in
commit fe82bb9b921be34370e6b71a1c6f062c20999ae0. the man pages
document that the getservby*_r functions set this pointer to
NULL if there was an error or if no record was found.
---
 src/network/getservbyname_r.c | 2 ++
 src/network/getservbyport_r.c | 1 +
 2 files changed, 3 insertions(+)

diff --git a/src/network/getservbyname_r.c b/src/network/getservbyname_r.c
index 056c2f3..ad3d616 100644
--- a/src/network/getservbyname_r.c
+++ b/src/network/getservbyname_r.c
@@ -15,6 +15,8 @@ int getservbyname_r(const char *name, const char *prots,
 	struct service servs[MAXSERVS];
 	int cnt, proto, align;
 
+	*res = 0;
+
 	/* Align buffer */
 	align = -(uintptr_t)buf & ALIGN-1;
 	if (buflen < 2*sizeof(char *)+align)
diff --git a/src/network/getservbyport_r.c b/src/network/getservbyport_r.c
index a0a7cae..0ae0e41 100644
--- a/src/network/getservbyport_r.c
+++ b/src/network/getservbyport_r.c
@@ -20,6 +20,7 @@ int getservbyport_r(int port, const char *prots,
 		if (r) r = getservbyport_r(port, "udp", se, buf, buflen, res);
 		return r;
 	}
+	*res = 0;
 
 	/* Align buffer */
 	i = (uintptr_t)buf & sizeof(char *)-1;
-- 
2.10.1

